aws/resource_aws_cloudsearch_domain.go

```diff
@@ -0,0 +1,486 @@
+package aws
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudsearch"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCloudSearchDomain() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCloudSearchDomainCreate,
+		Read:   resourceAwsCloudSearchDomainRead,
+		Update: resourceAwsCloudSearchDomainUpdate,
+		Delete: resourceAwsCloudSearchDomainDelete,
+
+		Schema: map[string]*schema.Schema{
+			"domain_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateDomainNameRegex,
+			},
+			"instance_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"replication_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  0,
+			},
+			"partition_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1,
+			},
+			"indexes": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"search": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						"facet": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"return": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						"sort": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"highlight": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"analysis_scheme": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"default_value": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"access_policy": {
+				Type:             schema.TypeString,
+				ValidateFunc:     validateIAMPolicyJson,
+				Required:         true,
+				DiffSuppressFunc: suppressEquivalentAwsPolicyDiffs,
+			},
```

Comment:
We are lacking a few attributes to retrieve when creating the CloudSearch domain:
* ARN
* DomainID
* SearchInstanceCount
* SearchInstanceType
* SearchPartitionCount
* SearchService Endpoint
* DocService Endpoint
* Limits

Also, some attributes here can be set as Computed, as values can be retrieved from the API

What do you think?